lib/deb/s3/lock.rb

```diff
@@ -52,7 +103,13 @@ def current(codename, component = nil, architecture = nil, cache_control = nil)
 
     private
     def lock_path(codename, component = nil, architecture = nil, cache_control = nil)
-      "dists/#{codename}/#{component}/binary-#{architecture}/lockfile"
+      #
+      # Acquire repository lock at `codename` level to avoid race between concurrent upload attempts.
+      #
+      # * `deb-s3 upload --arch=all` touchs multiples of `dists/{codename}/{component}/binary-*/Packages*`
+      # * All `deb-s3 upload` touchs `dists/{codename}/Release`
+      #
+      "dists/#{codename}/lockfile"
```

Comment:
I ended up to change the location of `lockfile` for safety sake. However, at the same time this brings interoperability issue with older releases of `deb-s3`. In case we should consider compatibility with older releases, I can tweak the lock logic to work better for that sake.